packages/types/Calendar.d.ts

```diff
@@ -120,12 +120,9 @@ export interface RecurringEvent {
   tzid?: string | undefined;
 }
 
-export interface IntervalLimit {
-  PER_DAY?: number | undefined;
-  PER_WEEK?: number | undefined;
-  PER_MONTH?: number | undefined;
-  PER_YEAR?: number | undefined;
-}
+export type IntervalLimitUnit = "day" | "week" | "month" | "year";
+
+export type IntervalLimit = Partial<Record<`PER_${Uppercase<IntervalLimitUnit>}`, number | undefined>>;
```

Comment:
abstracted limit units as they were repeatedly asserted in many different places